src/main/java/com/ericsson/ei/subscription/SubscriptionRepeatDbHandler.java

```diff
@@ -82,70 +86,72 @@
 
-        if (checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(subscriptionId, requirementId, aggrObjId)) {
-            LOGGER.debug("Subscription: " + subscriptionId + " and AggrObjId, " + aggrObjId
-                    + " has already been matched." + "No need to register the subscription match.");
-            return;
-        }
+            if (checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(subscriptionId, requirementId, aggrObjId)) {
+                LOGGER.debug("Subscription: " + subscriptionId + " and AggrObjId, " + aggrObjId
+                        + " has already been matched." + "No need to register the subscription match.");
+                return;
+            }
 
-        if (!updateExistingMatchedSubscriptionWithAggrObjId(subscriptionId, requirementId, aggrObjId)) {
-            LOGGER.error("Couldn't update SubscriptionMatched id.");
-            LOGGER.debug(
-                    "New Subscription AggrId has not matched, inserting new SubscriptionId and AggrObjId to matched list.");
-            BasicDBObject document = new BasicDBObject();
-            document.put("subscriptionId", subscriptionId);
+            if (!updateExistingMatchedSubscriptionWithAggrObjId(subscriptionId, requirementId, aggrObjId)) {
+                LOGGER.error("Couldn't update SubscriptionMatched id.");
+                LOGGER.debug(
+                        "New Subscription AggrId has not matched, inserting new SubscriptionId and AggrObjId to matched list.");
+                BasicDBObject document = new BasicDBObject();
+                document.put("subscriptionId", subscriptionId);
 
-            ArrayList<String> aggrObjIdsList = new ArrayList<String>();
-            aggrObjIdsList.add(aggrObjId);
+                ArrayList<String> aggrObjIdsList = new ArrayList<String>();
+                aggrObjIdsList.add(aggrObjId);
 
-            BasicDBObject reqDocument = new BasicDBObject();
-            reqDocument.put(String.valueOf(requirementId), aggrObjIdsList);
+                BasicDBObject reqDocument = new BasicDBObject();
+                reqDocument.put(String.valueOf(requirementId), aggrObjIdsList);
 
-            document.put("requirements", reqDocument);
+                document.put("requirements", reqDocument);
 
-            LOGGER.debug("New Matched AggrIdObject update on Subscription to be inserted to Db: " + document);
-            boolean result = mongoDbHandler.insertDocument(dataBaseName, collectionName, document.toString());
-            if (result == false) {
-                throw new Exception("Failed to insert the document into database");
+                LOGGER.debug("New Matched AggrIdObject update on Subscription to be inserted to Db: " + document);
+                mongoDbHandler.insertDocument(dataBaseName, collectionName, document.toString());
             }
+        } catch (MongoWriteException e) {
+            LOGGER.error("Failed to insert the document into database");
```

Comment:
Log exception together with the log message, so the exception and the log message is not spread out in the logs.

Reply:
Will fix